Examples/test-suite/cpp17_map_no_default_ctor.i

```diff
@@ -0,0 +1,15 @@
+%module cpp17_map_no_default_ctor
+// Tests c++17 insert_or_assign for std::map with no default constructor
+
+%include <std_map.i>
+
+%inline %{
+struct NoDefaultConstructorStruct final
+{
+	int value{0};
+	NoDefaultConstructorStruct(int v) : value(v) {}
+};
+using NoDefaultConstructorStructMap = std::map<int, NoDefaultConstructorStruct>;
+%}
+
+%template(NoDefaultConstructorStructMap) std::map<int, NoDefaultConstructorStruct>;

```

Comment:
Github unhelpfully doesn't show it, but I noticed looking locally that this new file seems to have DOS line endings.

Reply:
Very interesting as all my editors are configured to use LF only :)
I don't remember how I created this file, I'm pretty sure it was a copy of another existing one. I'll check this and commit a fix.

Regarding python, as I have almost no knowledge of it, what do you want me to do? Should I just push a new commit with your suggested change to std::pair, or do you want to do it separately as it affects pair and not map?

Reply:
It affects `std::map` too because `Lib/python/std_map.i` uses `std::pair`.  However I'm not sure what the full consequences of the patch I showed above are so I'm hoping wsfulton will get a chance to comment.